src/v/kafka/server/group_manager.cc

```diff
@@ -222,38 +222,47 @@ ss::future<> group_manager::handle_partition_leader_change(
          * we just became leader. make sure the log is up-to-date. see
          * struct group_log_record_key{} for more details.
          */
-        return inject_noop(p->partition, timeout).then([this, timeout, p] {
```

Comment:
adding a comment here becuase github doesn't allow for commit comments. 

Can you make a comment on how often this lock is contended. If it is not very contended, then it should be fine. if it its actively contended, I'd have to think about the interactions a bit deeper.